### ISSUE TITLE
Saiku Adhoc with BI Server 5

### ISSUE THREAD

**Issue by Dzo67** [Mgiepz/saiku-adhoc#178]
Recently I've done some testing with a snapshot build of BI Server 5. (I need that version to get my prpt reports working that are made with PRD 4 in order to get the crosstabs right)

So I also tested the Saiku Ad hoc plugin with BI Server 5, and it is not visible after logging on. To the BI Server snapshot I have added the system/cda subdirectory and the system/saiku-adhoc subdirecory, with their contents, as the ctools installer script says.

Do you know if Saiku Adhoc will work with BI Server 5?

In the Catalina.out file I see:
misc-class org.pentaho.platform.plugin.services.pluginmgr.DefaultPluginManager: PluginManager.ERROR_0015 - Plugin cda loaded, but post-load processing failed
java.lang.NoSuchMethodError: pt.webdetails.cda.CdaBoot.createDefaultHierarchicalConfiguration(Ljava/lang/String;Ljava/lang/String;ZLjava/lang/Class;)Lorg/pentaho/reporting/libraries/base/config/Configuration;

but some lines before that:

SEVERE: The web application [/pentaho] created a ThreadLocal with key of type [null](value [org.pentaho.platform.plugin.services.pluginmgr.PluginMessageLogger$1@db3925]) and a value of type [java.util.ArrayList](value [[Discovered plugin reporting in reporting, Content type registered: prpt - Pentaho Report, This plugin provides: 1 menu customizations, 0 content types, 0 content generators, 1 overlays, and {4} lifecycle listeners, Discovered plugin common-ui in common-ui, This plugin provides: 0 menu customizations, 2 content types, 0 content generators, 0 overlays, and {4} lifecycle listeners, Discovered plugin cgg in cgg, This plugin provides: 0 menu customizations, 1 content types, 0 content generators, 0 overlays, and {4} lifecycle listeners, Discovered plugin test-plugin-perspective in test-plugin-perspective, This plugin provides: 0 menu customizations, 0 content types, 0 content generators, 0 overlays, and {4} lifecycle listeners, Discovered plugin data-access in data-access, This plugin provides: 0 menu customizations, 0 content types, 1 content generators, 0 overlays, and {4} lifecycle listeners, Discovered plugin url in url, Content type registered: url - URL, This plugin provides: 1 menu customizations, 0 content types, 0 content generators, 0 overlays, and {4} lifecycle listeners, Discovered plugin BI Server default plugin in default-plugin, This plugin provides: 0 menu customizations, 3 content types, 1 content generators, 0 overlays, and {4} lifecycle listeners, Discovered plugin pdi-platform-plugin in pentaho-pdi-platform-plugin, Content type registered: ktr - Kettle Transformation, Content type registered: kjb - Kettle Job, This plugin provides: 2 menu customizations, 0 content types, 0 content generators, 1 overlays, and {4} lifecycle listeners, Discovered plugin pentaho-cdf in pentaho-cdf, Content type registered: xcdf - Dashboard, This plugin provides: 1 menu customizations, 0 content types, 0 content generators, 1 overlays, and {4} lifecycle listeners, Discovered plugin Saiku-Adhoc in saiku-adhoc, Content type registered: adhoc - Saiku Reporting, This plugin provides: 1 menu customizations, 2 content types, 1 content generators, 0 overlays, and {4} lifecycle listeners, Discovered plugin waqr in waqr, Content type registered: xwaqr - Ad-Hoc report, Content generator not registered: xwaqr.editor for plugin waqr, This plugin provides: 1 menu customizations, 0 content types, 2 content generators, 0 overlays, and {4} lifecycle listeners, Discovered plugin xaction in xaction, Content type registered: xaction - Action Sequence, This plugin provides: 1 menu customizations, 0 content types, 0 content generators, 0 overlays, and {4} lifecycle listeners, Plugin reporting has registered successfully, Content generator registered: reposvc for plugin common-ui, Content generator registered: MetadataModelsSvc for plugin common-ui, Plugin common-ui has registered successfully, Content generator registered: cgg for plugin cgg, Plugin cgg has registered successfully, Plugin test-plugin-perspective has registered successfully, Plugin data-access has registered successfully, Plugin url has registered successfully, Content generator registered: ws-services for plugin BI Server default plugin, Content generator registered: ws-run for plugin BI Server default plugin, Content generator registered: ws-wsdl for plugin BI Server default plugin, Plugin BI Server default plugin has registered successfully, Plugin pdi-platform-plugin has registered successfully, Plugin pentaho-cdf has registered successfully, Content generator registered: adhoc-service for plugin Saiku-Adhoc, Content generator registered: saiku-adhoc for plugin Saiku-Adhoc, Plugin Saiku-Adhoc has registered successfully, Plugin waqr has registered successfully, Plugin xaction has registered successfully]]) but failed to remove it when the web application was stopped. This is very likely to create a memory leak.

**Comment by Mgiepz** [comment]
it probably wont work because a lot of api changed. i am quite sure that 
the cda.jar that is used in saiku-adhoc is not compatible with reporting 
engine 4

Am 08.02.2013 10:24, schrieb Dzo67:

**Comment by Dzo67** [comment]
Does that mean that for now a combination of PRD4 prpt files and Saiku ad hoc is not possible in the same BI Server?
If so, do you know what developments are to be expected?

**Comment by Mgiepz** [comment]
correct. saiku-reporting 2 will exclusively use prpt-4 and therefor be 
only available in platform 5. this means we're making a cut here

Am 08.02.2013 12:54, schrieb Dzo67:

**Comment by Dzo67** [comment]
So the inevitable question is: 
Can you give an Estimated Time of Availability of saiku-reporting 2? 

Is saiku-reporting the version that combines the adhoc reports with analytical (MDX?) cubes?

**Comment by Mgiepz** [comment]
Am 08.02.2013 15:13, schrieb Dzo67:
